tee-worker/service/src/main_impl.rs

```diff
@@ -193,18 +193,16 @@ pub(crate) fn main() {
 
 		// litentry: start the mock-server if enabled
 		if config.enable_mock_server {
-			if_development_or!(
-				{
-					let mock_server_port = config
-						.try_parse_mock_server_port()
-						.expect("mock server port to be a valid port number");
-					thread::spawn(move || {
-						info!("*** Starting mock server");
-						let _ = lc_mock_server::run(mock_server_port);
-					});
-				},
-				warn!("Mock server not started. Node is running in production mode.")
-			)
+			#[cfg(any(feature = "mock-server", feature = "development"))]
```

Comment:
Since it's `any` here, why don't we use `development` feature only to control it? Do we have a case where we want to launch mock-server for production build?

Additionally, it could be a bit misleading if we have `WORKER_MOCK_SERVER=0` and `WORKER_DEV=1` => the mock-server will be launched (even if we set it explicitly to 0)

Reply:
We cannot use only `development` feature in this case because we want to run CI checks with mock server while having build without `development` feature.


Two conditions needs to be met in order to start mock server:
1. worker needs to be built with one of mentioned features
2. `config.enable_mock_server` must be set to true

I agree that it can be misleading that worker built without `mock-server` feature will launch it if config allows for that - but I assumed it's still fine since there is `development` feature. I can change it so it will be included only by `mock-server` feature, what do you think?